backend/src/main/java/com/empire/World.java

```diff
@@ -1190,14 +1185,14 @@ public double sum() {
 							region.setReligion(ct.religion, this);
 							if (r != Ideology.VESSEL_OF_FAITH && region.religion == Ideology.VESSEL_OF_FAITH) {
 								for (String kingdom : kingdoms.keySet()) if (Ideology.VESSEL_OF_FAITH == NationData.getStateReligion(kingdom, this)) {
-									for (Region rr : regions) if (kingdom.equals(rr.kingdom)) rr.unrestPopular = Math.max(0, rr.unrestPopular - .1);
+									for (Region rr : regions) if (kingdom.equals(rr.getKingdom())) rr.unrestPopular = Math.max(0, rr.unrestPopular - .1);
 								}
 							}
 							if (getNation(c.kingdom).hasTag("Mystical")) region.unrestPopular = Math.max(0, region.unrestPopular - .1);
 							if (ct.religion.religion != Religion.IRUHAN) getNation(c.kingdom).goodwill -= 20;
 							else if (ct.religion != Ideology.VESSEL_OF_FAITH) getNation(c.kingdom).goodwill += 15;
 						}
-						if (ct.type.equals("shipyard") && getNation(c.kingdom).hasTag("Ship-Building")) buildShips(c.kingdom, c.location, 10);
+						if (ct.type.equals("shipyard") && getNation(c.kingdom).hasTag("Ship-Building")) buildShips(c.kingdom, c.location, Constants.numShipsBuiltPerShipyard * Constants.shipBuildingTraitWeeksProduction);
```

Comment:
Might need an updated explanation for ship-building, this has a new shipyard building 25 ships the first turn.

Reply:
I think the existing in-game explanation is sufficiently vague - of course, maybe 25 warships is not "small". (Was 10?)